ticdc: remove canal protocol document (#16773)
pingcap/docs: ticdc/ticdc-faq.md

- Replicates all tables except system tables
- Only replicates tables that contain [valid indexes](/ticdc/ticdc-overview.md#best-practices)

## Does TiCDC support outputting data changes in the Canal protocol?

> Yes. Note that for Canal protocol, TiCDC only support the JSON output format, while the protobuf format is not officially supported yet. To enable Canal output, specify the protocol as `canal-json` in the `--sink-uri` parameter. For example:

Reviewer: ```suggestion
Yes. Note that for the Canal protocol, TiCDC only supports the JSON output format, while the protobuf format is not officially supported yet. To enable Canal output, specify `protocol` as `canal-json` in the `--sink-uri` configuration. For example:
```

> ## Does TiCDC support outputting data changes in the Canal format?

Reviewer 2: ```suggestion
## Does TiCDC support outputting data changes in the Canal protocol?
```

> Yes. To enable Canal output, specify the protocol as `canal-json` in the `--sink-uri` parameter. For example:

Reviewer 2: ```suggestion
Yes. Note that for Canal protocol, TiCDC only support the JSON output format, while the protobuf format is not officially supported yet. To enable Canal output, specify the protocol as `canal-json` in the `--sink-uri` parameter. For example:
```

{{< copyable "shell-regular" >}}

```shell
cdc cli changefeed create --server=http://127.0.0.1:8300 --sink-uri="kafka://127.0.0.1:9092/cdc-test?kafka-version=2.4.0&protocol=canal-json" --config changefeed.toml
```

> **Note:**
>
> * This feature is introduced in TiCDC 4.0.2.
> * TiCDC currently supports outputting data changes in the Canal-JSON format only to MQ sinks such as Kafka.

> > * The difference between Canal and Canal-JSON is that Canal uses protobuf to do the encoding, while Canal-JSON uses JSON.

Reviewer 2: ```suggestion
```
We simply don't mention `Canal` at all.
